make the Video HDR item three-state when "10-bit color" is on

Smali changes of vendor_oplus_camera d1d3ce0, over baksmali --sequential-labels output.

diff --git a/classes7/gq/h.smali b/classes7/gq/h.smali
index 39b20d44149348204196aca10cb72a0ff9b80d36..ef41b3d4b7db153658ba15b1b5584993f1d1f800 100644
--- a/classes7/gq/h.smali
+++ b/classes7/gq/h.smali
@@ -11781,6 +11781,14 @@
 .method public final Y(Ljava/lang/String;Z)V
     .locals 7
 
+    invoke-virtual {p0}, Lgq/h;->s()Ljava/lang/String;
+
+    move-result-object v0
+
+    invoke-static {v0, p1, p2}, Lmm/DV;->click(Ljava/lang/String;Ljava/lang/String;Z)Ljava/lang/String;
+
+    move-result-object p1
+
     .line 1
     iget-object v0, p0, Lgq/h;->j:Leo/b;
 
@@ -11798,7 +11806,7 @@
 
     .line 8
     .line 9
-    if-ltz v1, :cond_17b
+    if-ltz v1, :cond_183
 
     .line 10
     .line 11
@@ -11849,7 +11857,7 @@
 
     .line 33
     .line 34
-    if-eqz v1, :cond_3f
+    if-eqz v1, :cond_47
 
     .line 35
     .line 36
@@ -11899,11 +11907,11 @@
     .line 60
     .line 61
     .line 62
-    goto/16 :goto_180
+    goto/16 :goto_188
 
     .line 63
     .line 64
-    :cond_3f
+    :cond_47
     invoke-virtual {v0}, Leo/b;->getOptionKey()Ljava/lang/String;
 
     .line 65
@@ -11923,7 +11931,7 @@
     const/4 v6, 0x4
 
     .line 73
-    if-eqz v1, :cond_60
+    if-eqz v1, :cond_68
 
     .line 74
     .line 75
@@ -11963,11 +11971,11 @@
     .line 93
     .line 94
     .line 95
-    goto/16 :goto_180
+    goto/16 :goto_188
 
     .line 96
     .line 97
-    :cond_60
+    :cond_68
     invoke-virtual {v0}, Leo/b;->getOptionKey()Ljava/lang/String;
 
     .line 98
@@ -11984,7 +11992,7 @@
     move-result v1
 
     .line 105
-    if-nez v1, :cond_79
+    if-nez v1, :cond_81
 
     .line 106
     .line 107
@@ -12008,7 +12016,7 @@
     move-result v1
 
     .line 117
-    if-nez v1, :cond_79
+    if-nez v1, :cond_81
 
     .line 118
     .line 119
@@ -12017,7 +12025,7 @@
     .line 120
     .line 121
     .line 122
-    :cond_79
+    :cond_81
     new-instance v1, Lvc/o;
 
     .line 123
@@ -12074,7 +12082,7 @@
     move-result v1
 
     .line 152
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 153
     .line 154
@@ -12098,7 +12106,7 @@
     move-result v1
 
     .line 164
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 165
     .line 166
@@ -12122,7 +12130,7 @@
     move-result v1
 
     .line 176
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 177
     .line 178
@@ -12146,7 +12154,7 @@
     move-result v1
 
     .line 188
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 189
     .line 190
@@ -12170,7 +12178,7 @@
     move-result v1
 
     .line 200
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 201
     .line 202
@@ -12194,7 +12202,7 @@
     move-result v1
 
     .line 212
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 213
     .line 214
@@ -12218,7 +12226,7 @@
     move-result v1
 
     .line 224
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 225
     .line 226
@@ -12242,7 +12250,7 @@
     move-result v1
 
     .line 236
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 237
     .line 238
@@ -12266,7 +12274,7 @@
     move-result v1
 
     .line 248
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 249
     .line 250
@@ -12290,7 +12298,7 @@
     move-result v1
 
     .line 260
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 261
     .line 262
@@ -12314,7 +12322,7 @@
     move-result v1
 
     .line 272
-    if-nez v1, :cond_11d
+    if-nez v1, :cond_125
 
     .line 273
     .line 274
@@ -12338,11 +12346,11 @@
     move-result v1
 
     .line 284
-    if-eqz v1, :cond_129
+    if-eqz v1, :cond_131
 
     .line 285
     .line 286
-    :cond_11d
+    :cond_125
     const-string v1, "pref_camera_200m"
 
     .line 287
@@ -12363,11 +12371,11 @@
     move-result v1
 
     .line 296
-    if-eqz v1, :cond_177
+    if-eqz v1, :cond_17f
 
     .line 297
     .line 298
-    :cond_129
+    :cond_131
     invoke-virtual {v0}, Leo/b;->getOptionKey()Ljava/lang/String;
 
     .line 299
@@ -12396,7 +12404,7 @@
     move-result v1
 
     .line 312
-    if-nez v1, :cond_155
+    if-nez v1, :cond_15d
 
     .line 313
     .line 314
@@ -12412,7 +12420,7 @@
     move-result v1
 
     .line 320
-    if-nez v1, :cond_155
+    if-nez v1, :cond_15d
 
     .line 321
     .line 322
@@ -12428,7 +12436,7 @@
     move-result v1
 
     .line 328
-    if-nez v1, :cond_155
+    if-nez v1, :cond_15d
 
     .line 329
     .line 330
@@ -12452,11 +12460,11 @@
     move-result v1
 
     .line 340
-    if-eqz v1, :cond_165
+    if-eqz v1, :cond_16d
 
     .line 341
     .line 342
-    :cond_155
+    :cond_15d
     iget-object v1, p0, Lgq/h;->w:Lcom/oplus/camera/q6;
 
     .line 343
@@ -12489,7 +12497,7 @@
     move-result-object v0
 
     .line 358
-    :cond_165
+    :cond_16d
     invoke-static {}, Lcom/oplus/camera/data/DataManager;->getInstance()Lcom/oplus/camera/data/DataManager;
 
     .line 359
@@ -12522,16 +12530,16 @@
     .line 374
     .line 375
     .line 376
-    :cond_177
+    :cond_17f
     invoke-virtual {p0}, Lgq/h;->g()V
 
     .line 377
     .line 378
     .line 379
-    goto :goto_180
+    goto :goto_188
 
     .line 380
-    :cond_17b
+    :cond_183
     const-string p0, "setOptionValue fail! index: "
 
     .line 381
@@ -12541,7 +12549,7 @@
     .line 383
     .line 384
     .line 385
-    :goto_180
+    :goto_188
     return-void
     .line 386
     .line 387
diff --git a/classes7/mm/DV.smali b/classes7/mm/DV.smali
index 56f1d36ac61bfb8d49da17338953c80ebd0a4a49..69d530f8af92768d2404b268ab3d4d49e4fc08b6 100644
--- a/classes7/mm/DV.smali
+++ b/classes7/mm/DV.smali
@@ -2,10 +2,25 @@
 .super Ljava/lang/Object;
 .source "DV.java"
 
+# interfaces
+.implements Ljava/lang/Runnable;
+
 
 # static fields
 .field static final RPU:[B
 
+.field static act:Ljava/lang/ref/WeakReference;
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "Ljava/lang/ref/WeakReference<",
+            "Landroid/app/Activity;",
+            ">;"
+        }
+    .end annotation
+.end field
+
+.field static volatile dolby:Z
+
 .field static last:Z
 
 .field static seen:Z
@@ -443,6 +458,102 @@
     return-void
 .end method
 
+.method public static click(Ljava/lang/String;Ljava/lang/String;Z)Ljava/lang/String;
+    .locals 1
+
+    .line 42
+    const-string v0, "on"
+
+    if-eqz p2, :cond_3f
+
+    :try_start_4
+    const-string p2, "key_video_dolby"
+
+    invoke-virtual {p2, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_3f
+
+    invoke-static {}, Lom/t0;->tenBitPref()Z
+
+    move-result p0
+
+    if-nez p0, :cond_13
+
+    goto :goto_3f
+
+    .line 43
+    :cond_13
+    invoke-virtual {v0, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    const/4 p2, 0x0
+
+    if-eqz p0, :cond_24
+
+    sget-boolean p0, Lmm/DV;->dolby:Z
+
+    if-eqz p0, :cond_23
+
+    sput-boolean p2, Lmm/DV;->dolby:Z
+
+    invoke-static {}, Lmm/DV;->rebuild()V
+
+    :cond_23
+    return-object p1
+
+    .line 44
+    :cond_24
+    const-string p0, "off"
+
+    invoke-virtual {p0, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-nez p0, :cond_2d
+
+    return-object p1
+
+    .line 45
+    :cond_2d
+    sget-boolean p0, Lmm/DV;->dolby:Z
+
+    if-nez p0, :cond_32
+
+    const/4 p2, 0x1
+
+    :cond_32
+    sput-boolean p2, Lmm/DV;->dolby:Z
+
+    .line 46
+    invoke-static {}, Lmm/DV;->rebuild()V
+
+    .line 47
+    sget-boolean p0, Lmm/DV;->dolby:Z
+    :try_end_39
+    .catchall {:try_start_4 .. :try_end_39} :catchall_3d
+
+    if-eqz p0, :cond_3c
+
+    move-object p1, v0
+
+    :cond_3c
+    return-object p1
+
+    .line 48
+    :catchall_3d
+    move-exception p0
+
+    return-object p1
+
+    .line 42
+    :cond_3f
+    :goto_3f
+    return-object p1
+.end method
+
 .method static dv(Lmm/i;)Z
     .locals 1
 
@@ -473,10 +584,10 @@
 .method public static in(Lmm/i;)V
     .locals 3
 
-    .line 41
+    .line 74
     iget-object v0, p0, Lmm/i;->h0:Landroid/media/MediaFormat;
 
-    .line 42
+    .line 75
     if-eqz v0, :cond_28
 
     invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
@@ -487,7 +598,7 @@
 
     goto :goto_28
 
-    .line 43
+    .line 76
     :cond_b
     const-string p0, "profile"
 
@@ -495,34 +606,34 @@
 
     invoke-virtual {v0, p0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 44
+    .line 77
     const-string p0, "level"
 
     invoke-virtual {v0, p0}, Landroid/media/MediaFormat;->removeKey(Ljava/lang/String;)V
 
-    .line 45
+    .line 78
     const-string p0, "color-standard"
 
     const/4 v2, 0x6
 
     invoke-virtual {v0, p0, v2}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 46
+    .line 79
     const-string p0, "color-transfer"
 
     const/4 v2, 0x7
 
     invoke-virtual {v0, p0, v2}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 47
+    .line 80
     const-string p0, "color-range"
 
     invoke-virtual {v0, p0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 48
+    .line 81
     return-void
 
-    .line 42
+    .line 75
     :cond_28
     :goto_28
     return-void
@@ -531,7 +642,7 @@
 .method public static out(Lmm/i;Landroid/media/MediaFormat;)V
     .locals 3
 
-    .line 51
+    .line 84
     invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
 
     move-result v0
@@ -554,51 +665,87 @@
 
     goto :goto_29
 
-    .line 52
+    .line 85
     :cond_15
     const-string v0, "video/dolby-vision"
 
     invoke-virtual {p1, v1, v0}, Landroid/media/MediaFormat;->setString(Ljava/lang/String;Ljava/lang/String;)V
 
-    .line 53
+    .line 86
     const-string v0, "profile"
 
     const/16 v1, 0x100
 
     invoke-virtual {p1, v0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 54
+    .line 87
     const-string v0, "level"
 
     iget p0, p0, Lmm/i;->g1:I
 
     invoke-virtual {p1, v0, p0}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 55
+    .line 88
     return-void
 
-    .line 51
+    .line 84
     :cond_29
     :goto_29
     return-void
 .end method
 
+.method static rebuild()V
+    .locals 2
+
+    .line 54
+    invoke-static {}, Lmm/DV;->tenBit()Z
+
+    move-result v0
+
+    sput-boolean v0, Lmm/DV;->last:Z
+
+    .line 55
+    new-instance v0, Landroid/os/Handler;
+
+    invoke-static {}, Landroid/os/Looper;->getMainLooper()Landroid/os/Looper;
+
+    move-result-object v1
+
+    invoke-direct {v0, v1}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
+
+    new-instance v1, Lmm/DV;
+
+    invoke-direct {v1}, Lmm/DV;-><init>()V
+
+    invoke-virtual {v0, v1}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
+
+    .line 56
+    return-void
+.end method
+
 .method public static resume(Landroid/app/Activity;)V
     .locals 2
 
-    .line 35
+    .line 67
+    new-instance v0, Ljava/lang/ref/WeakReference;
+
+    invoke-direct {v0, p0}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
+
+    sput-object v0, Lmm/DV;->act:Ljava/lang/ref/WeakReference;
+
+    .line 68
     invoke-static {}, Lmm/DV;->tenBit()Z
 
     move-result v0
 
-    .line 36
+    .line 69
     sget-boolean v1, Lmm/DV;->seen:Z
 
-    if-eqz v1, :cond_12
+    if-eqz v1, :cond_19
 
     sget-boolean v1, Lmm/DV;->last:Z
 
-    if-eq v0, v1, :cond_12
+    if-eq v0, v1, :cond_19
 
     sput-boolean v0, Lmm/DV;->last:Z
 
@@ -606,22 +753,22 @@
 
     return-void
 
-    .line 37
-    :cond_12
+    .line 70
+    :cond_19
     const/4 p0, 0x1
 
     sput-boolean p0, Lmm/DV;->seen:Z
 
     sput-boolean v0, Lmm/DV;->last:Z
 
-    .line 38
+    .line 71
     return-void
 .end method
 
 .method public static sample(Lmm/i;ILjava/nio/ByteBuffer;Landroid/media/MediaCodec$BufferInfo;)Ljava/nio/ByteBuffer;
     .locals 6
 
-    .line 58
+    .line 91
     invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
 
     move-result v0
@@ -644,7 +791,7 @@
 
     goto :goto_4c
 
-    .line 59
+    .line 92
     :cond_15
     iget p0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
 
@@ -658,12 +805,12 @@
 
     move-result-object p0
 
-    .line 60
+    .line 93
     invoke-virtual {p2}, Ljava/nio/ByteBuffer;->duplicate()Ljava/nio/ByteBuffer;
 
     move-result-object p1
 
-    .line 61
+    .line 94
     iget p2, p3, Landroid/media/MediaCodec$BufferInfo;->offset:I
 
     iget v0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
@@ -678,7 +825,7 @@
 
     invoke-virtual {p2, v0}, Ljava/nio/ByteBuffer;->position(I)Ljava/nio/ByteBuffer;
 
-    .line 62
+    .line 95
     invoke-virtual {p0, p1}, Ljava/nio/ByteBuffer;->put(Ljava/nio/ByteBuffer;)Ljava/nio/ByteBuffer;
 
     move-result-object p1
@@ -691,7 +838,7 @@
 
     invoke-virtual {p1}, Ljava/nio/ByteBuffer;->flip()Ljava/nio/ByteBuffer;
 
-    .line 63
+    .line 96
     invoke-virtual {p0}, Ljava/nio/ByteBuffer;->limit()I
 
     move-result v2
@@ -706,32 +853,44 @@
 
     invoke-virtual/range {v0 .. v5}, Landroid/media/MediaCodec$BufferInfo;->set(IIJI)V
 
-    .line 64
+    .line 97
     return-object p0
 
-    .line 58
+    .line 91
     :cond_4c
     :goto_4c
     return-object p2
 .end method
 
 .method public static tenBit()Z
-    .locals 1
+    .locals 2
 
-    .line 28
-    :try_start_0
+    .line 35
+    const/4 v0, 0x1
+
+    :try_start_1
     invoke-static {}, Lom/t0;->tenBitPref()Z
 
-    move-result v0
-    :try_end_4
-    .catchall {:try_start_0 .. :try_end_4} :catchall_5
+    move-result v1
 
-    return v0
+    if-eqz v1, :cond_c
 
-    :catchall_5
-    move-exception v0
+    sget-boolean v1, Lmm/DV;->dolby:Z
+    :try_end_9
+    .catchall {:try_start_1 .. :try_end_9} :catchall_e
 
-    const/4 v0, 0x1
+    if-eqz v1, :cond_c
+
+    goto :goto_d
+
+    :cond_c
+    const/4 v0, 0x0
+
+    :goto_d
+    return v0
+
+    :catchall_e
+    move-exception v1
 
     return v0
 .end method
@@ -821,3 +980,43 @@
     :goto_3e
     return v1
 .end method
+
+
+# virtual methods
+.method public run()V
+    .locals 2
+
+    .line 59
+    sget-object v0, Lmm/DV;->act:Ljava/lang/ref/WeakReference;
+
+    if-nez v0, :cond_6
+
+    const/4 v0, 0x0
+
+    goto :goto_e
+
+    :cond_6
+    sget-object v0, Lmm/DV;->act:Ljava/lang/ref/WeakReference;
+
+    invoke-virtual {v0}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
+
+    move-result-object v0
+
+    check-cast v0, Landroid/app/Activity;
+
+    .line 60
+    :goto_e
+    if-eqz v0, :cond_19
+
+    invoke-virtual {v0}, Landroid/app/Activity;->isFinishing()Z
+
+    move-result v1
+
+    if-nez v1, :cond_19
+
+    invoke-virtual {v0}, Landroid/app/Activity;->recreate()V
+
+    .line 61
+    :cond_19
+    return-void
+.end method
